src/index.ts

```diff
@@ -148,7 +150,7 @@ export type { AuthMechanismId } from './cmap/auth/defaultAuthProviders';
 export type { MongoCredentials, MongoCredentialsOptions } from './cmap/auth/mongo_credentials';
 export type {
   WriteProtocolMessageType,
-  Query,
+  Query as __InternalQuery,
```

Comment:
This wireprotocol class Query conflicts with the new `Query<>` type. Completely open to whatever naming we like.
In the community types it was referred to as `FilterQuery<>` which is just the two terms we often used pushed together but it seems like we should strive to pick one? thoughts?

Reply:
In this case driver specifications have always used "filter" as the terminology post `OP_QUERY`, so my preference would be `Filter` here.

Reply:
> In this case driver specifications have always used "filter" as the terminology post `OP_QUERY`, so my preference would be `Filter` here.

Reply:
Going off the API examples in the MongoDB manual it seems we use it interchangably, I do think filter will eliminate this issue so let's go with that.

Reply:
leaving this conversation open to make sure we get rid of any outward facing instances of "query" in parameters and replace with "filter"